### PR TITLE
bump Android build-tools to 29.0.2, compileSdk to 29

### PR DESCRIPTION
**Pull request by dulmandakh**
## Summary

Bump bump Android build-tools to 29.0.2, compileSdk to 29, Buck to 2019.10.17.01.

## Changelog

[Android] [Changed] - bump Android build-tools to 29.0.2, compileSdk to 29

## Test Plan

CI is green

### CODE REVIEW

**Comment by dulmandakh** [comment]
changing targetSdk usually breaks many things, so this is a preparation
step.

**Comment by dulmandakh** [comment]
@mdvacca could you please share failure information with me. Please merge

**Comment by mdvacca** [comment]
It is failing because we still need to make some changes internally. I will keep you updated

**Comment by dulmandakh** [comment]
@hramos please review and merge

File: template/android/build.gradle [hunk 1]
```diff
@@ -2,9 +2,9 @@
 
 buildscript {
     ext {
-        buildToolsVersion = "28.0.3"
+        buildToolsVersion = "29.0.2"
         minSdkVersion = 16
-        compileSdkVersion = 28
+        compileSdkVersion = 29
         targetSdkVersion = 28
```
Comment (SaeedZhiany) [inline template/android/build.gradle]: @dulmandakh 

It seems you forgot to bump tagetSdkVersion(s) to 29 too. AFAIK `targetSdkVersion` should not be less than `compileSdkVersion`.

I read [the above comment](https://github.com/facebook/react-native/pull/26909#issuecomment-564411531), are you want to merge these changes before bumping `targetSdkVersion`

Comment (friederbluemle) [reply template/android/build.gradle]: @SaeedZhiany - If you check my (collapsed) comment from Oct 18, 2019 above, you can see I had the same question, and @dulmandakh already responded. It sounds like the update to `targetSdkVersion` will come at a later point.

Comment (SaeedZhiany) [reply template/android/build.gradle]: Yeah, I saw that and my question is why we don't wait until we can bump targetSdkVersion too, then merge it. usually Android studio shows a warning (or error) message on Gradle file when targetSdkVersion less than compileSdkVersion. (I can't remember what the message is though)

**Comment by mdvacca** [comment]
This is still blocked on an internal dependency, I'm expecting we will be able to land this next week

**Comment by dulmandakh** [comment]
landed with https://github.com/facebook/react-native/commit/edcbfb98210d9aaa6bb1d7c64281ae9cfb41cac2

**Pull request closed without merging**